core/app/models/spree/stock_location.rb

```diff
@@ -12,8 +12,8 @@ class StockLocation < Spree::Base
     scope :active, -> { where(active: true) }
     scope :order_default, -> { order(default: :desc, name: :asc) }
 
-    after_create :create_stock_items, :if => "self.propagate_all_variants?"
     after_save :ensure_one_default
+    after_commit :create_stock_items, on: :create, if: "self.try :propagate_all_variants?"
```

Comment:
Perhaps Spree 4.0 we need to revisit the migrations to move seed data out to where it should be, and insist migrating data be handled by upgrade rake tasks in the future.  At some point it probably wouldn't hurt to consolidate migrations again like we did for https://github.com/spree/spree/blob/master/core/db/migrate/20120831092320_spree_one_two.rb